stash_engine/config/routes.rb

```diff
@@ -58,6 +58,7 @@
   match 'auth/:provider/callback', :to => 'sessions#callback', :via => [:get, :post]
   match 'auth/migrate/mail', :to => 'dashboard#migrate_data_mail', :via => [:get]
   match 'auth/migrate/code', :to => 'dashboard#migrate_data', :via => [:get]
+  match 'auth/migrate/done', to: 'dashboard#migration_complete', via: [:get]
```

Comment:
just a style question: are the `=>` things not necessary, or does this do something slightly different without them?

Reply:
(Probably this could be enforced and even autofixed with RuboCop.)

Reply:
Like David says :x => 'value' is equivalent to x: 'value' in a hash.

The => symbol is the old way Ruby did things through Ruby 1.8 for hashes and people call it the "hash rocket."  I think people got annoyed typing it with the equals and shift period key.  Also JSON was the new hotness, so they wanted to look JSON-y.

The colon in the hash does two separate things which is it 1) makes the preceding thing into a symbol and 2) it separates key and value (the hash rocket only does the second of the two).

Rails uses symbols for keys most everywhere so you see a lot of it.  Rails is also is very un-picky about hashes vs strings as keys and will most of the time will retrieve either one by the other, so something like my_hash[:cat] and my_hash['cat'] for things like the params hash will return the same value.

It does get pickier about hashes you create yourself (or in plain Ruby), though.